test/dns/query.cpp

```diff
@@ -99,6 +104,18 @@ TEST_CASE("throw error while fails evdns_set_options for attempts") {
     REQUIRE(false);
 }
 
+TEST_CASE("throw error while fails evdns_set_options for timeout") {
+    try {
+        create_evdns_base<::evdns_base_new, ::evdns_base_nameserver_ip_add,
+                          base_free_evdns_set_options_timeout>(
+            {{"dns/attempts", "nexa"}}, Reactor::global());
+    } catch (std::runtime_error &) {
+        REQUIRE(base_free_evdns_set_options_timeout_flag);
+        return;
+    }
```

Comment:
Yes, that was clear to me. I am just asking whether you could try rewriting that piece of code using Catch's native support for checking whether code excepts, as opposed to rolling out an handcrafted piece of code doing that functionality.